timesketch/frontend/src/components/Sigma/SigmaList.vue

```diff
@@ -127,14 +204,165 @@ export default {
+          })
+          .catch(e => {
+          })
+      }
+      if (this.save_button_text === "Update") {
+        // Only update the rule if the parsing was positive.
+        ApiClient.getSigmaRuleByText(this.editingRule.rule_yaml)
+          .then(response => {
+            let SigmaRule = response.data.objects[0]
+            this.parsed = SigmaRule
+            ApiClient.updateSigmaRule(this.editingRule.id, this.editingRule.rule_yaml)
+              .then(response => { })
+              .catch(e => {
+                console.error(e)
+              })
+            location.reload();
+            this.$buefy.notification.open({ message: 'Succesfully modified Sigma rule!', type: 'is-success' })
+            this.showEditModal = false
+            this.data_type_present = (SigmaRule['search_query'].includes("data_type") || SigmaRule['search_query'].includes("source_name"))
+          })
+          .catch(e => {
+          })
+      }
+      //
+    },
+    composeRule() {
+      this.showEditModal = true
+      this.create_new_rule = true
+      this.save_button_text = "Create"
+      this.editingRule = { rule_yaml: this.text }
+    },
+    startRuleEdit(rule) {
+      if (rule === null)
+        console.log("No rule given, maybe you wanna add one?")
+      this.save_button_text = "Update"
+      this.showEditModal = true
+      this.editingRule = rule
+    },
+    deleteRule(ioc) {
+      if (confirm('Delete Rule?')) {
+        ApiClient.deleteSigmaRule(ioc.rule_uuid)
+          .then(response => { })
+          .catch(e => {
+            console.error(e)
+          })
+        // TODO: remove the element from the array
+      }
+      // instead of removing the element from the array gonna relad the page
+      location.reload();
+    },
+    generateOpenSearchQuery(value, field) {
+      let query = `"${value}"`
+      // Escape special OpenSearch characters: \, [space]
+      query = query.replace(/[\\\s]/g, '\\$&')
+      if (field !== undefined) {
+        query = `${field}: ${query}`
+      }
+
+      return { q: query }
+    },
+    loadSketchSigmaTags() {
+      ApiClient.runAggregator(this.sketch.id, {
+        aggregator_name: 'field_bucket',
+        aggregator_parameters: { field: 'ts_sigma_rule' },
+      }).then((response) => {
+        this.sketchTags = response.data.objects[0].field_bucket.buckets
+        // of the form [{count: 0, tag: 'foo'}]
+      })
+    },
+    loadSketchTTP() {
+      ApiClient.runAggregator(this.sketch.id, {
+        aggregator_name: 'field_bucket',
+        aggregator_parameters: { field: 'ts_ttp' },
+      }).then((response) => {
+        this.sketchTTP = response.data.objects[0].field_bucket.buckets
+        // of the form [{count: 0, tag: 'foo'}]
+      })
+    },
+    getColor(status) {
+      if (status === 'false') return "red"
+      return "green";
+    },
+    getRuleByUUID(RuleUuid) {
```

Comment:
```suggestion
    getRuleByUUID(ruleUuid) {
```

camelCase starts with lowercase